scripts/gen_vimdoc.py

```diff
@@ -32,24 +33,29 @@
     parameter is marked as [out].
   - Each function documentation is separated by a single line.
 """
+
+from __future__ import annotations
```

Comment:
Do we need to pull this in? Can we not upgrade the min python version instead? Otherwise can we add a comment justifying why we need this.

Reply:
nvm, I see you added this so you don't need to use `Union`.

Reply:
Yes, I intended min python version to be 3.7 (was 3.6 before), so I believe this shouldn't be a big problem. But there's actually one small mistake I missed: `typing.Literal` requires 3.8, so actually it ended up being 3.8+. (We can relax to 3.7 by getting rid of `import Literal`)

Reply:
I can recommend "vermin" to quickly determine minimum python version in script.